client/component/HorizontalAxisSelectionConfig.tsx

```diff
@@ -0,0 +1,55 @@
+import { LabelledInput } from './LabelledInput';
+import { getSelectionLabel, SELECTION_ICONS } from './selections/utils';
+import { isNumber, toFixedNumber } from './utils';
+import HorizontalAxisSelection from './selections/HorizontalAxisSelection';
+import {
+  AlphaInput,
+  ColourPicker,
+  NameInput,
+  XInput,
+} from './SelectionConfigComponents';
+
+interface HorizontalAxisSelectionConfigProps {
+  selection: HorizontalAxisSelection;
+  updateSelections: (s: SelectionBase | null, b?: boolean, c?: boolean) => void;
+}
+
+export function HorizontalAxisSelectionConfig(
+  props: HorizontalAxisSelectionConfigProps
+) {
+  return (
+    <>
+      <h4 key="Selection">
+        {' '}
+        {getSelectionLabel(props.selection, SELECTION_ICONS)}{' '}
+      </h4>
+      <ColourPicker
+        selection={props.selection}
+        updateSelections={props.updateSelections}
+      />
+      <NameInput
+        selection={props.selection}
+        updateSelections={props.updateSelections}
+      />
+
+      <AlphaInput
+        selection={props.selection}
+        updateSelections={props.updateSelections}
+      />
```

Comment:
Put this and above together in SelectionConfig as they are common to all configs

Reply:
Just the function definitions or push to `modeless` like with `ClearSelectionBtn`?

Reply:
Like `ClearSelectionBtn`. There, in SelectionConfig, you can just test for `currentSelection` once rather than repeatedly.